content/en/blog/_posts/2021-04-22-gateway-api/2021-04-22-gateway-api.md

```diff
@@ -0,0 +1,199 @@
+
+
+We discussed the first sketches for Gateway API in the halls of Kubecon 2019 San Diego (and overflowing to a hotel lobby). The concept of evolving the Ingress API was based on a couple key assumptions:
+
+1. The API standards underlying route matching, traffic management, and service exposure are commoditized and provide little value to their implementers and users as custom APIs
+2. It’s possible to represent L4/L7 routing, traffic management through common core API resources
+3. It’s possible to provide extensibility for more complex capabilities in a way that does not sacrifice the user experience of the core API
+
+
+## Introducing the Gateway API
+
+This led to design principles that allow the Gateway API to improve upon Ingress:
+
+- **Expressiveness** - In addition to HTTP host/path matching and TLS, Gateway API can express capabilities like HTTP header manipulation, traffic weighting & mirroring, TCP/UDP routing, and other capabilities that were only possible in Ingress through custom annotations.
+- **Role-oriented design** - The API resource model reflects the separation of responsibilities that is common in routing and Kubernetes service networking.
+- **Extensibility** - The resources allow arbitrary configuration attachment at various layers within the API. This makes granular customization possible at the most appropriate places.
+- **Flexible conformance** - This is where the Gateway API aims to improve the portability Ingress through more [flexible conformance](https://gateway-api.sigs.k8s.io/concepts/guidelines/#conformance). With core (mandatory support), extended (portable if supported), and custom (no portability guarantee) conformance levels, the API is explicit about its portability without constraining implementations-specific capabilities.
+
+
+### What does the Gateway API look like?
+
+The Gateway API introduces a few new resource types:
+
+- **GatewayClasses** are a cluster-scoped resource that act as templates to explicitly define behavior for Gateways derived from them. This is similar in concept to StorageClasses, but for Gateways, our Ingress-like object.
+- **Gateways** are the deployed instances of GatewayClasses. They are the logical representation of the data-plane which performs routing, which may be in-cluster proxies, hardware LBs, or cloud LBs.
+- **HTTPRoutes** are the matching, filtering, and routing rules that get applied to Gateways, which results in the configuration of the underlying data-plane. There are also other kinds of Routes that exist too (since they are protocol-specific).
+- **xRoutes** are not a resource itself, but represent that Routes are protocol-specific. This provides a path for future evolution.  TLSRoutes, TCPRoutes, and UDPRoutes already have proposals with the future possibility of other protocols such as GRPCRoutes.
```

Comment:
```suggestion
- **xRoutes** are not a resource itself, but represent that Routes are protocol-specific. This provides a path for future evolution. TLSRoutes, TCPRoutes, and UDPRoutes are already included in the core API and implementations are encouraged to create their own Route resources.
```

Reply:
+1, that's more correct.

Reply:
Have updated this to:

**Routes** represent many different protocol-specific Route resources. These include HTTPRoutes, TCPRoutes, and UDPRoutes which have protocol-specific routing semantics. This also allows the Gateway API to expand its protocol support in the future.